src/sentry/api_gateway/proxy.py

```diff
@@ -0,0 +1,54 @@
+"""
+Utilities related to proxying a request to a region silo
+"""
+
+from urllib.parse import urljoin
+
+import requests as external_request
+from requests import Response as external_response
+from rest_framework.exceptions import NotFound
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.utils import generate_region_url
+from sentry.models.organization import Organization
+
+
+def get_org_region(slug: str) -> str:
+    from sentry.types.region import get_region_for_organization
+
+    org = Organization.objects.get(slug=slug)
+    region_info = get_region_for_organization(org)
+    return region_info.address
+
+
+def build_url(region: str, request: Request) -> str:
+    """Might need some changes to handle other format region address."""
```

Comment:
I agree with the comment; the whole business around the region silo's URL is a little confusing.

There's already an issue with `src.sentry.api.utils.generate_region_url` and `Region.to_url` doing similar jobs (which is on @dashed and me to sync up). Part of the reason I was nitpicking about the word "utilities" in the docstring above is that it suggests that functions like this one are intended to be "published" as an interface for future code, which we don't want to do redundantly.

AFAICT, `proxy_request` is the only entry point to this module. I'd recommend prefixing all the other function names with an underscore.